attools/src/SwiftBuildTool.swift

```diff
@@ -13,71 +13,195 @@
-    will enforce that API on OSX as well */
-    private static let xcTestCaseProvider: String = { () -> String in
-        var s = ""
-        s += "import XCTest\n"
-        s += "\n"
-        s += "protocol XCTestCaseProvider {\n"
-        s += "    var allTests : [(String, () -> Void)] { get }\n"
-        s += "}\n"
-        s += "\n"
-        s += "public func XCTMain(testCases: [XCTestCase]) {\n"
-        s += "    fatalError(\"Can't get here.\")\n"
-        s += "}\n"
-        s += "\n"
-        s += "extension XCTestCase {\n"
-        s += "    private func implementAllTests() {\n"
-        s += "        print(\"Make sure to implement allTests via\")\n"
-        s += "        print(\"extension \\(self.dynamicType) : XCTestCaseProvider {\")\n"
-        s += "        print(\"    var allTests : [(String, () -> Void)] {\")\n"
-        s += "        print(\"        return [\")\n"
-        s += "        print(\"        (\\\"testFoo\\\", testFoo)\")\n"
-        s += "        print(\"        ]\")\n"
-        s += "        print(\"    }\")\n"
-        s += "        print(\"}\")\n"
-        s += "        print(\"(Or disable xctestStrict.)\")\n"
-        s += "        print(\"Cheers! -- Anarchy Tools Team\")\n"
-        s += "    }\n"
-        s += "    override public func tearDown() {\n"
-        s += "        if let provider = self as? XCTestCaseProvider {\n"
-        s += "            let contains = provider.allTests.contains({ test in\n"
-        s += "                return test.0 == invocation!.selector.description\n"
-        s += "            })\n"
-        s += "            if !contains {\n"
-        s += "               XCTFail(\"Test \\(name) is missing from \\(self.dynamicType)\")\n"
-        s += "               implementAllTests()\n"
-        s += "            }\n"
-        s += "        }\n"
-        s += "        else {\n"
-        s += "            XCTFail(\"Whoops!  \\(self.dynamicType) doesn't conform to XCTestCaseProvider.\")\n"
-        s += "            implementAllTests()\n"
-        s += "        }\n"
-        s += "        super.tearDown()\n"
-        s += "    }\n"
-        s += "}\n"
-        s += "\n"
-        return s
-    }()
+/**
+ * The `llbuild.swift` file contains all of the tools necessary to work
+ * with the swift-build-tool (llbuild) low-level build system. There are
+ * three tools contained here:
+ *
+ *    1. llbuild-config - this tool is reponsible for generate the build
+ *       configuration file necessary for llbuild to consume.
+ *    2. llbuild-build - this tool is responsible for consuming a build
+ *       configuration file and producing the appropriate outputs.
+ *    3. llbuild - this is a wrapper tool that provides a simple syntax
+ *       to merge the previous two tools together.
+ */
+
+/**
+ * This tool handles the generation of the configuration file for the
+ * llbuild tool (`swift-build-tool`).
+ */
+final class SwiftBuildToolConfig: Tool {
+    enum Keys {
```

Comment:
This is a minor thing, but I'd be interested in your argument for why not

```swift
enum Keys : String {
    case Tool = "tool"
    //etc.
}
```

I use that pattern a lot, if there is something wrong with it, it would be useful for me to know.